packages/terra-date-input/src/DateInputField.jsx

```diff
@@ -0,0 +1,240 @@
+  // eslint-disable-next-line react/forbid-prop-types
+  monthAttributes: PropTypes.object,
+  /**
+   * Set the max-width of a field using `length` or `%`.  Best practice recommendation to never exceed
+   * a rendered value of 1020px. _(Note: Providing custom inline styles will take precedence.)_
+   */
+  maxWidth: PropTypes.string,
+  /**
+   * A callback function triggered when the date input component loses focus.
+   * This event does not get triggered when the focus is moved from the date input to the calendar button since the focus is still within the main date input component.
+   */
+  onBlur: PropTypes.func,
+  /**
+   * A callback function to execute when a date is entered.
+   * The first parameter is the event. The second parameter is the changed date value.
+   */
+  onChange: PropTypes.func,
+  /**
+   * A callback function triggered when the date input component receives focus.
+   * This event does not get triggered when the focus is moved from the date input to the calendar button since the focus is still within the main date input component.
+   */
+  onFocus: PropTypes.func,
+  /**
+   * Whether or not the date input field is required.
+   */
+  required: PropTypes.bool,
+  /**
+   * Whether or not to append the 'optional' label to a non-required field label.
+   */
+  showOptional: PropTypes.bool,
+  /**
+   * An date string representation of the date value used for the component. This should be in ISO 8601 format: YYYY-MM-DD.
+   */
+  value: PropTypes.string,
+  /**
+   * Custom input attributes to apply to the year input
+   */
+  // eslint-disable-next-line react/forbid-prop-types
+  yearAttributes: PropTypes.object,
+};
+
+const defaultProps = {
+  dayAttributes: {},
+  disabled: false,
+  displayFormat: undefined,
+  error: null,
+  help: null,
+  hideRequired: false,
+  isInline: false,
+  isInvalid: false,
+  isLegendHidden: false,
+  legendAttributes: {},
+  monthAttributes: {},
+  maxWidth: undefined,
+  onBlur: undefined,
+  onChange: undefined,
+  onFocus: undefined,
+  required: false,
+  showOptional: false,
+  value: '',
+  yearAttributes: {},
+};
+
+const DateInputField = (props) => {
+  const {
+    dayAttributes,
+    disabled,
+    displayFormat,
+    error,
+    help,
+    hideRequired,
+    isInline,
+    isInvalid,
+    isLegendHidden,
+    legend,
+    legendAttributes,
+    monthAttributes,
+    maxWidth,
+    name,
+    onBlur,
+    onChange,
+    onFocus,
+    required,
+    showOptional,
+    value,
+    yearAttributes,
+    ...customProps
+  } = props;
+
+  const dateInputFieldClasses = cx([
+    'date-input-field',
+    { 'is-inline': isInline },
+    customProps.className,
+  ]);
+
+  const legendClassNames = cx([
+    'legend',
+    legendAttributes.className,
+  ]);
+
+  /**
+   * aria-describedby ids
+   * Used to map legend, help text, and error messages with input/select elements
+   */
+  const helpAriaDescriptionId = help ? `terra-date-input-field-description-help-${uniqueid()}` : '';
```

Comment:
Is uniqueId going to change with each render? Does that matter?

Reply:
Yeah, it looks like it is changing with each render. I'll update this to use uuid like we do in [terra-navigation-prompt](https://github.com/cerner/terra-framework/blob/7ec7a5589cb2dc17a0520d2fd4bd469b30c45f31/packages/terra-navigation-prompt/src/NavigationPrompt.jsx#L33) instead so its stays the same through renders.

Users don't need to interact with this ID, but we do want it to be unique.

Reply:
Updated to use uuid for unique ids in this commit: https://github.com/cerner/terra-framework/pull/838/commits/0549f6fb6ed0373bd06c8b80d005a2a983d4ccc1